src/main/java/ch/njol/skript/lang/function/DynamicFunctionReference.java

```diff
@@ -174,12 +175,20 @@ else if (parameters.length < signature.getMinParameters())
 
 		// Check parameter types
 		for (int i = 0; i < parameters.length; i++) {
-			Parameter<?> parameter = signature.parameters[varArgs ? 0 : i];
+			Parameter<?> parameter = signature.parameters().values().toArray(new Parameter<?>[0])[varArgs ? 0 : i];
```

Comment:
given the existing need to get parameters by order, I think you should provide some help for that in Signature. Perhaps consider a different data structure for parameters or keeping a separate ordered list of params by name as a lookup.